README.md

```diff
@@ -314,6 +286,50 @@ If we want to run a command on all search results, we can pipe the output to `xa
 Here, the `-0` option tells *fd* to separate search results by the NULL character (instead of
 newlines). In the same way, the `-0` option of `xargs` tells it to read the input in this way.
 
+### Parallel command execution
+If the `-x`/`--exec` option is specified alongside a command template, a job pool will be created
+for executing commands in parallel for each discovered path as the input. The number of threads 
+used for command execution can be set with the `--threads`/`-j` option.
+
+*fd* takes the command template as a series of arguments rather than as a string. If you want to 
+add additional options after the command template, you can terminate it with a `\;`. This is 
+useful when you want to repeat a command with new options. For example, to remove write and 
+execute permissions from all directories, run:
+``` bash
+fd -t d -x chmod -wx
+```
+If you realize you also need to modify hidden directories, you can quickly add the `-H` (or `--hidden`) 
+option after the command template:
+```bash
+fd -t d -x chmod -wx \; -H
+```
+
+More examples:
+``` bash
+# Convert all jpg files to png files:
+fd -e jpg -x convert {} {.}.png
+
+# Unpack all zip files (if no placeholder is given, the path is appended):
+fd -e zip -x unzip
+
+# Convert all flac files into opus files:
+fd -e flac -x ffmpeg -i {} -c:a libopus {.}.opus
+
+# Count the number of lines in Rust files (the command template can be terminated with ';'):
+fd -x wc -l \; -e rs
+```
+
+The syntax for generating commands is similar to that of GNU Parallel:
+
+- `{}`: A placeholder token that will be replaced with the path of the search result
+  (`documents/images/party.jpg`).
+- `{.}`: Like `{}`, but without the file extension (`documents/images/party`).
+- `{/}`: A placeholder that will be replaced by the basename of the search result (`party.jpg`).
+- `{//}`: Uses the parent of the discovered path (`documents/images`).
+- `{/.}`: Uses the basename, with the extension removed (`party`).
+
+If you do not include a placeholder, *fd* automatically adds `{}`.
```

Comment:
This is fine here, but maybe we should explain above what will actually be executed if you run `fd -e zip -x unzip` or `fd -e zip -X unzip`.

Reply:
Honestly, I didn't even think of the --exec-batch option. Should that be included in the Parallel command execution section, or would it be better to have that as a separate section?

Reply:
Also, when you "say maybe we should explain above what will actually be executed" do you mean in the existing block of examples? After the examples? Or just above the line "If you do not include a placeholder, *fd* automatically adds `{}`." Sorry, I'm not quite sure what you have in mind.

Reply:
I guess I was wondering if you meant something a bit like this; a short example which explicitly says which commands are run.

--------------
If the `-x`/`--exec` option is specified alongside a command template, a job pool will be created
for executing commands in parallel for each discovered path as the input.

In the following example, `fd` searches the current directory recursively for any file with the 
`zip` extension, and it runs `unzip` separately for each search result.
```bash
fd -e zip -x unzip
```
Thus, if there are two `zip` files, `file1.zip` and `example-dir/file2.zip`, the example runs `unzip file1.zip`
and `unzip example-dir/file2.zip`.

More Examples:
``` bash
# Convert all jpg files to png files:
fd -e jpg -x convert {} {.}.png

# Unpack all zip files (if no placeholder is given, the path is appended):
fd -e zip -x unzip

# Convert all flac files into opus files:
fd -e flac -x ffmpeg -i {} -c:a libopus {.}.opus

# Count the number of lines in Rust files (the command template can be terminated with ';'):
fd -x wc -l \; -e rs
```

Reply:
> Honestly, I didn't even think of the --exec-batch option. Should that be included in the Parallel command execution section, or would it be better to have that as a separate section?

To me, the `-X`/`--exec-batch` option is so useful that it should be discussed in the same section, I think. We can simply rename it to "Command execution" maybe, because with `-X` there is no parallelism.

> I guess I was wondering if you meant something a bit like this; a short example which explicitly says which commands are run.

Yes, exactly! 

> Thus, if there are two `zip` files, `file1.zip` and `example-dir/file2.zip`, the example runs `unzip file1.zip`
> and `unzip example-dir/file2.zip`.

:+1: Maybe we could add: "The two `unzip` processes run in parallel (if the files are found fast enough)."